packages/kubernetes-1.17/kubelet.service

```diff
@@ -15,7 +15,8 @@ ExecStartPre=/usr/bin/host-ctr \
     --containerd-socket=/run/dockershim.sock \
     --namespace=k8s.io \
     pull-image \
-    --source=${POD_INFRA_CONTAINER_IMAGE}
+    --source=${POD_INFRA_CONTAINER_IMAGE} \
+    --registry-config=/etc/host-containers/host-ctr-registry.config
```

Comment:
It's OK for this file to be empty? Also, it is sharing the host-ctr's registry config? It makes sense that the setting would be the same for host-ctr and kubelet, but is it weird for kubelet to depend on something in `/etc/host-containers`?

Reply:
If this file is empty, then the unmarshalled list of mirrors would be empty. So the image would just be pulled from the host as is.

The reason why this is here is because the pause container needs to be pulled and cached before kubelet starts up. We've depended on host-ctr to do that. The registry mirrors would affect that process as well.
It is sharing the same registry config. This isn't something that can be set in kubelet's config file. All container management is done by the containerd runtime. kubelet has no hand in this.